src/lib/commandclass/ColorSwitchCC.ts

```diff
@@ -0,0 +1,762 @@
+
+		// Refresh the current value
+		await this.get(propertyKey);
+	};
+}
+
+@commandClass(CommandClasses["Color Switch"])
+@implementedVersion(3)
+export class ColorSwitchCC extends CommandClass {
+	declare ccCommand: ColorSwitchCommand;
+
+	public async interview(complete: boolean = true): Promise<void> {
+		const node = this.getNode()!;
+		const endpoint = this.getEndpoint()!;
+		const api = endpoint.commandClasses["Color Switch"];
+
+		log.controller.logNode(node.id, {
+			endpoint: this.endpointIndex,
+			message: `${this.constructor.name}: doing a ${
+				complete ? "complete" : "partial"
+			} interview...`,
+			direction: "none",
+		});
+
+		// TODO: Only do this if complete?  Where would we get
+		//	the previously discovered colors?
+		log.controller.logNode(node.id, {
+			endpoint: this.endpointIndex,
+			message: "querying Color Switch CC supported colors...",
+			direction: "outbound",
+		});
+
+		const colorsResponse = await api.getSupported();
+
+		log.controller.logNode(node.id, {
+			endpoint: this.endpointIndex,
+			message: "querying Color Switch CC color states...",
+			direction: "outbound",
+		});
+
+		for (const key of ColorKeys) {
+			const supportsKey = supportedColorTableKey(key);
+			if (!colorsResponse[supportsKey]) {
+				continue;
+			}
+
+			await api.get(key);
+		}
+	}
+}
+
+@CCCommand(ColorSwitchCommand.SupportedReport)
+export class ColorSwitchCCSupportedReport extends ColorSwitchCC
+	implements Partial<SupportedColorTable> {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		// Docs say 'variable length', but the table shows 2 bytes.
+		validatePayload(this.payload.length >= 2);
+
+		this._supportedColors = {
+			supportsWarmWhite: Boolean(this.payload[0] & 1),
+			supportsColdWhite: Boolean(this.payload[0] & 2),
+			supportsRed: Boolean(this.payload[0] & 4),
+			supportsGreen: Boolean(this.payload[0] & 8),
+			supportsBlue: Boolean(this.payload[0] & 16),
+			supportsAmber: Boolean(this.payload[0] & 32),
+			supportsCyan: Boolean(this.payload[0] & 64),
+			supportsPurple: Boolean(this.payload[0] & 128),
+			supportsIndex: Boolean(this.payload[1] & 1),
+		};
+
+		const node = this.getNode()!;
+		const endpoint = this.getEndpoint()!;
+
+		for (const key of ColorKeys) {
+			const supportsColorKey = supportedColorTableKey(key);
+
+			const valueId: ValueID = {
+				commandClass: CommandClasses["Color Switch"],
+				endpoint: endpoint.index,
+				property: "supportedColor",
+				propertyKey: key,
+			};
+
+			// TODO: Should we be setting this every run?
+			node.valueDB.setMetadata(valueId, {
+				...ValueMetadata.ReadOnly,
+				label: `Supports ${key}`,
+				description: `Whether the endpoint supports setting the ${key} color.`,
+			});
+			node.valueDB.setValue(
+				valueId,
+				this._supportedColors[supportsColorKey] ?? false,
+			);
+		}
+	}
+
+	private _supportedColors: SupportedColorTable | null;
```

Comment:
This is definitely assigned in the constructor, so null should not be necessary.

Reply:
And you can remove the `| undefined` and `?.` from the properties below.